### PR TITLE
Fix timezone issue with autogen tracing

### PR DESCRIPTION
**Pull request by B-Step62**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/B-Step62/mlflow/pull/13047?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/13047/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 13047
```

</p>
</details>

### What changes are proposed in this pull request?

During running autogen tracing in my own laptop, I found that the span duration for LLM call is not correct (about 32400s). The root cause was that the timestamp obtained [here](https://github.com/mlflow/mlflow/blob/f810d03ec4188c06c61219650b97e64df7748aa1/mlflow/autogen/autogen_logger.py#L236-L238) does not handle the local timezone correctly and added 9 hours offset in my case. This was not caught in manual testing before because our staging account is based on UTC timezone.

Screenshot 2024-09-02 at 19 16 02

### How is this PR tested?

- [x] Existing unit/integration tests
- [x] New unit/integration tests
- [x] Manual tests

Screenshot 2024-09-02 at 19 15 34

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Fix timezone handling in the AutoGen tracing integration.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [x] Yes (this PR will be cherry-picked and included in the next patch release)
- [ ] No (this PR will be included in the next minor release)

### CODE REVIEW

File: tests/autogen/test_autogen_autolog.py [hunk 1]
```diff
@@ -251,6 +251,33 @@ def sum(a: int, b: int) -> int:
     assert tool_span.end_time_ns - tool_span.start_time_ns >= 1e9  # 1 second
 
 
+def test_tracing_llm_completion_duration_timezone(llm_config, monkeypatch):
+    # Test if the duration calculation for LLM completion is robust to timezone changes.
+    monkeypatch.setenv("TZ", "Asia/Tokyo")
+    time.tzset()
```
Comment (harupy) [inline tests/autogen/test_autogen_autolog.py:257]: can we reset the time zone when the test finishes?

**Review by daniellok-db — APPROVED** [review]
lgtm!

File: tests/autogen/test_autogen_autolog.py (hunk 1 above)
Comment (B-Step62) [reply tests/autogen/test_autogen_autolog.py:257]: ah sure will do!

**Review by harupy — APPROVED** [review]
LGTM!

**Pull request merged**